lib/TableGen/Operations.cpp

```diff
@@ -64,11 +84,24 @@ bool OperationBase::init(raw_ostream &errs, GenDialectsContext &context,
   auto arguments = parseArguments(errs, context, record);
   if (!arguments.has_value())
     return false;
+
   m_arguments = std::move(*arguments);
 
   for (const auto &arg : m_arguments) {
     std::string attrType;
 
+    if (!m_hasVariadicArgument && arg.type->isVarArgList()) {
+      if (m_superclass && m_superclass->hasVariadicArgument()) {
+        errs << "Superclass " << m_superclass->name
+             << " already has variadic arguments, cannot set "
+                "again on "
+             << arg.name << ".\n";
+        return false;
+      }
+
+      m_hasVariadicArgument = true;
+    }
+
```

Comment:
If the superclass ends with varargs, then having *any* argument is not okay. The check should mostly be moved outside the loop; if the superclass has varargs, check that there aren't *any* arguments here, and also set this->m_hasVariadicArgument.

Reply:
Ah, I see, this is already handled by `SymbolTable::chooseName` - it seems to iteratively lookup the symbol table for either "args" or "arguments" and either chooses the other name or uses an incremental suffix. Cool! :)